incorrect archetype name in java functions quickstart guide

navicore opened the pull request:
fixing error:
```
The desired archetype does not exist (com.microsoft.azure:azure-functions-archetype:1.0-SNAPSHOT)
```

azurecla commented:
Hi __@arunvinoth__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

navicore committed:
incorrect archetype name

fixing error:
```
The desired archetype does not exist (com.microsoft.azure:azure-functions-archetype:1.0-SNAPSHOT)
```
